doc/plugins.rst

```diff
@@ -28,6 +28,36 @@ set of artifacts, for example::
 The ``url`` item should be set to the file path of the artifacts, and the
 message is what will be shown in the status check.
 
+Push handlers
+-------------
+
+We provide a plugin that will perform custom actions whenever a push is made to
```

Comment:
Some examples of what constitutes a "push" would be nice. For instance, push made directly to a branch (`master` or otherwise) or merge of a PR? What about pushing tags; does that count? And maybe specify how one could ask the handler to only do something on one kind of push but not another. If all these are documented on GitHub somewhere, then provide a link instead.